### ISSUE TITLE
Make `PubSubSink` public

### ISSUE THREAD

**Issue by uima-chen** [redis-rs/redis-rs#1365]
Hi! I ask for this because I want to make a wrapper type for `SubPub` splitted, basically like:

```rust
use anyhow::{Context, Result};
use futures::Stream;

struct Wrapper {
    sink : redis::aio::PubSubSink,
}

impl Wrapper {
    async fn new(redis_client: &redis::Client) -> Result<(Self, impl Stream)> {
        let (sink, stream) = redis_client
            .get_async_pubsub()
            .await
            .context("failed to get pubsub")?
            .split();

        let wrapper= Self { sink };

        Ok((wrapper, stream))
    }

    async fn subscribe(channel: &str) -> Result<()> {
        self.sink.subscribe(channel).await.context("failed to subscribe")?
        Ok(())
    }
}
```

But since `SubPubSink` is private type, so I cannot use that in my wrapper type, though I can get that by `SubPub::split()`.

I have learned that I can use `impl Stream` for `SubPubStream`, so the stream part is fine, but I can't find the way to do with the sink part. Do I miss something or this is a bad practice?

**Comment by uima-chen** [comment]
haha, Thanks for the quick response and fix! :)

**Comment by nihohit** [comment]
fixed in 0.27.4
